internal/controller/operator/factory/vmanomaly/deploy.go

```diff
@@ -0,0 +1,194 @@
+package vmanomaly
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	vmv1 "github.com/VictoriaMetrics/operator/api/operator/v1"
+	vmv1beta1 "github.com/VictoriaMetrics/operator/api/operator/v1beta1"
+	"github.com/VictoriaMetrics/operator/internal/controller/operator/factory/build"
+	"github.com/VictoriaMetrics/operator/internal/controller/operator/factory/finalize"
+	"github.com/VictoriaMetrics/operator/internal/controller/operator/factory/reconcile"
+)
+
+// CreateOrUpdate creates vmanomalyand and builds config for it
+func CreateOrUpdate(ctx context.Context, cr *vmv1.VMAnomaly, rclient client.Client) error {
+	var prevCR *vmv1.VMAnomaly
+	if cr.ParsedLastAppliedSpec != nil {
+		prevCR = cr.DeepCopy()
+		prevCR.Spec = *cr.ParsedLastAppliedSpec
+	}
+	if err := deletePrevStateResources(ctx, rclient, cr, prevCR); err != nil {
+		return fmt.Errorf("cannot delete objects from prev state: %w", err)
+	}
+	if cr.IsOwnsServiceAccount() {
+		var prevSA *corev1.ServiceAccount
+		if prevCR != nil {
+			prevSA = build.ServiceAccount(prevCR)
+		}
+		if err := reconcile.ServiceAccount(ctx, rclient, build.ServiceAccount(cr), prevSA); err != nil {
+			return fmt.Errorf("failed create service account: %w", err)
+		}
+		if ptr.Deref(cr.Spec.UseVMConfigReloader, false) {
```

Comment:
I think, config reloader is not needed, this configuration hash is written into annotations and it will trigger restart of application each time configuration is changed.